Sources/web3swift/Transaction/EIP1559Envelope.swift

```diff
@@ -0,0 +1,320 @@
+                guard let itemData = accessData[index] else { return nil }
+                guard let newItem = AccessListEntry(rlpItem: itemData)  else { return nil }
+                newList.append(newItem)
+            }
+            self.accessList = newList
+        }
+    }
+
+    public init(to: EthereumAddress, nonce: BigUInt? = nil,
+                chainID: BigUInt? = nil, value: BigUInt? = nil, data: Data,
+                v: BigUInt = 1, r: BigUInt = 0, s: BigUInt = 0,
+                options: TransactionOptions? = nil) {
+        self.to = to
+        self.nonce = nonce ?? options?.resolveNonce(0) ?? 0
+        self.internalChainID = chainID ?? 0
+        self.value = value ?? options?.value ?? 0
+        self.data = data
+        self.v = v
+        self.r = r
+        self.s = s
+        // decode gas options, if present
+        self.maxPriorityFeePerGas = options?.resolveMaxPriorityFeePerGas(0) ?? 0
+        self.maxFeePerGas = options?.resolveMaxFeePerGas(0) ?? 0
+        self.gasLimit = options?.resolveGasLimit(0) ?? 0
+        // get the access list, if present
+        self.accessList = options?.accessList ?? []
+    }
+
+    // memberwise
+    public init(to: EthereumAddress, nonce: BigUInt = 0,
+                chainID: BigUInt = 0, value: BigUInt = 0, data: Data,
+                maxPriorityFeePerGas: BigUInt = 0, maxFeePerGas: BigUInt = 0, gasLimit: BigUInt = 0,
+                accessList: [AccessListEntry]? = nil,
+                v: BigUInt = 1, r: BigUInt = 0, s: BigUInt = 0) {
+        self.to = to
+        self.nonce = nonce
+        self.internalChainID = chainID
+        self.value = value
+        self.data = data
+        self.maxPriorityFeePerGas = maxPriorityFeePerGas
+        self.maxFeePerGas = maxFeePerGas
+        self.gasLimit = gasLimit
+        self.accessList = accessList ?? []
+        self.v = v
+        self.r = r
+        self.s = s
+    }
+
+    public mutating func applyOptions(_ options: TransactionOptions) {
+        // type cannot be changed here, and is ignored
+        self.nonce = options.resolveNonce(self.nonce)
+        self.maxPriorityFeePerGas = options.resolveMaxPriorityFeePerGas(self.maxPriorityFeePerGas)
+        self.maxFeePerGas = options.resolveMaxFeePerGas(self.maxFeePerGas)
+        self.gasLimit = options.resolveGasLimit(self.gasLimit)
+        self.value = options.value ?? self.value
+        self.to = options.to ?? self.to
+        self.accessList = options.accessList ?? self.accessList
+    }
+
+    public func getOptions() -> TransactionOptions {
+        var options = TransactionOptions()
+        options.nonce = .manual(self.nonce)
+        options.maxPriorityFeePerGas = .manual(self.maxPriorityFeePerGas)
+        options.maxFeePerGas = .manual(self.maxFeePerGas)
+        options.gasLimit = .manual(self.gasLimit)
+        options.value = self.nonce
+        options.to = self.to
+        options.accessList = self.accessList
+        return options
+    }
```

Comment:
I had thought about that, and it certainly works either way. I elected not to do that, as options may not alter all possible values, depending on it's contents. I know it's semantics, but I felt that because options is not a 1:1 computed vs stored structure it was best left as a method.

Reply:
I do see some utility though, as it would allow for `envelope.options.value = value` which I could then shim within EthereumTransaction to get `transaction.options.value = value` 

I'm starting to feel convinced on this :)

Reply:
Further thoughts... if we go the computed property route, perhaps it's better to drop using `TransactionOptions` here, and create a new struct called `TransactionParameters` or something like that. It would be similar to `TransactionOptions`, but would not include the policy stuff, it would just be a combined list of the parameters of all transaction types in optional form. This would also remove any confusion created by the type name having "options" in it.

This would potentially move `TransactionOptions` back to `Web3` but that's okay, as right now it does double-duty between the two levels. 

thoughts?